ngx-tools/utilities/src/abbreviate-number/abbreviate-number.ts

```diff
@@ -0,0 +1,41 @@
+
+/**
+ * Helper function to abbreviate a number
+ *
+ * @param input - The number to be abbreviated.
+ * @param decimalPlace - The decimals users define for final abbreviated number. Default to 1.
+ * @return The abbreviated number
+ *
+ * @example
+ * abbreviateNumber(1234, '1')              // Returns: '1.2K'
+ */
+export function abbreviateNumber(input: number | null, decimalPlace = 1): string {
+  const SCALE_NUMBER = 3;
+  const MATH_POWER = 10;
+
+  if (!input) {
+    return '';
+  }
+  const exponential = input
+    .toExponential()
+    .split('e+')
+    .map(el => +el);
+
+  if (exponential[1] < SCALE_NUMBER) {
+    return input.toString();
+  }
+
+  const modulus = exponential[1] % SCALE_NUMBER;
+  exponential[0] = exponential[0] * Math.pow(MATH_POWER, modulus);
```

Comment:
I see that these words are not abbreviated anymore, but it hasn't really solved the core of the problem.

The reason we have chosen to not rely on abbreviations is because it can make code less clear for humans. It seems as though this latest change hasn't actually helped on that front. 

The `exponential` and `modulus` variables are easier to read as a human, but aren't they now misrepresented? The `exponential` var actually has an array stored and the `modulus` is storing a number.

I think we should take the time to really name things well since we are building this for human consumers.

Reply:
I somehow misunderstood your comments. Thought you just wanted the names to be spelled out, instead of abbreviated.